galley/pkg/config/analysis/analyzers/annotations/annotations.go

```diff
@@ -226,12 +167,10 @@ func lookupAnnotation(ann string) *annotation.Instance {
 }
 
 func resourceTypesAsStrings(resourceTypes []annotation.ResourceTypes) []string {
-	retval := make([]string, len(resourceTypes))
-	var ok bool
-	for i, resourceType := range resourceTypes {
-		retval[i], ok = resourceTypeNames[resourceType]
-		if !ok {
-			retval[i] = "Unknown"
+	retval := []string{}
+	for _, resourceType := range resourceTypes {
+		if s := resourceType.String(); s != "Unknown" {
```

Comment:
Are you intentionally changing the behavior here? Before it was adding "Unknown" and now its not

Reply:
Yes. If we are adding `Unknown` here, the output string in the analyzer message will contain `Unknown` which is misleading.

Reply:
resourceType.String() should never return `Unknown` so there is no point in filtering it from the result.  This analyzer was written before I had a good way to get a string from a ResourceType.